src/service/data/modelmesh_parser.py

```diff
@@ -0,0 +1,293 @@
+        if param_name not in tensor.parameters:
+            return False
+
+        param = tensor.parameters[param_name]
+        return hasattr(param, "string_param") and param.string_param == "true"
+
+    @staticmethod
+    def _get_tensor_data(tensor, tensor_type: str, request_obj: Any) -> np.ndarray:
+        """Extract data from a tensor."""
+        if ModelMeshPayloadParser._has_tensor_data(tensor):
+            return ModelMeshPayloadParser._extract_tensor_data(tensor, tensor_type)
+        elif (
+            hasattr(request_obj, "raw_input_contents")
+            and request_obj.raw_input_contents
+        ):
+            raise NotImplementedError("Raw input contents parsing not yet implemented")
+        elif (
+            hasattr(request_obj, "raw_output_contents")
+            and request_obj.raw_output_contents
+        ):
+            raise NotImplementedError("Raw output contents parsing not yet implemented")
+        else:
+            raise ValueError(f"No data found in tensor {tensor.name}")
+
+    @staticmethod
+    def _extract_output_tensors(
+        output_response: ModelInferResponse,
+    ) -> Tuple[List[np.ndarray], List[str]]:
+        """Extract features and names from output response."""
+        output_features = []
+        output_names = []
+
+        for tensor in output_response.outputs:
+            tensor_type = ModelMeshPayloadParser._get_tensor_type(tensor.datatype)
+            shape = list(tensor.shape)
+
+            data = ModelMeshPayloadParser._get_tensor_data(
+                tensor, tensor_type, output_response
+            )
+
+            if len(shape) > 1:
+                data = data.reshape(shape)
+
+            output_features.append(data)
+            output_names.append(tensor.name)
+
+        return output_features, output_names
+
+    @staticmethod
+    def _build_dataframe_rows(
+        input_features: List[np.ndarray],
+        input_names: List[str],
+        output_features: List[np.ndarray],
+        output_names: List[str],
+        synthetic_flags: List[bool],
+        id: str,
+        model_id: str,
+        batch_size: int,
+    ) -> List[Dict[str, Any]]:
+        """Build DataFrame rows from extracted features."""
+        rows = []
+        for i in range(batch_size):
+            row = {}
+
+            # Inputs
+            for j, name in enumerate(input_names):
+                data = input_features[j]
+                row[name] = data[i] if i < len(data) else None
+
+            # Outputs
+            for j, name in enumerate(output_names):
+                data = output_features[j]
+                row[f"output_{name}"] = data[i] if i < len(data) else None
+
+            row["id"] = id
+            row["model_id"] = ModelMeshPayloadParser.standardize_model_id(model_id)
+            row["synthetic"] = any(synthetic_flags)
+
+            rows.append(row)
+
+        return rows
+
+    @staticmethod
+    def payloads_to_dataframe(
```

Comment:
Are these just for debugging?

Reply:
The reconciliation is done via a temporary dataframe for ModelMesh payloads.
If you're ok with it, we can change this in the next PR, to bring more in line with the KServe reconciliation.